internal/standalone/standalone.go

```diff
@@ -17,113 +18,188 @@ import (
+	standaloneAddress, err := getStandaloneAddress(pluginID, dir)
 	if err != nil {
-		return Args{}, err
+		return ClientSettings{}, err
+	}
+
+	if standaloneAddress == "" {
+		return ClientSettings{}, fmt.Errorf("address is empty")
 	}
 
-	return getInfo(id, ex, standalone, debug)
+	standalonePID, err := getStandalonePID(dir)
+	if err != nil {
+		return ClientSettings{}, err
+	}
+
+	return ClientSettings{
+		TargetAddress: standaloneAddress,
+		TargetPID:     standalonePID,
+	}, nil
 }
 
-// will check a few options to find the dist plugin json file
-func findPluginJSON(exe string) (string, error) {
+func currentProcPath() (string, error) {
+	curProcPath, err := os.Executable()
+	if err != nil {
+		return "", err
+	}
+	return curProcPath, nil
+}
+
+// findPluginRootDir will attempt to find a plugin directory based on the executing process's file-system path.
+func findPluginRootDir(curProcPath string) (string, error) {
 	cwd, _ := os.Getwd()
 	if filepath.Base(cwd) == "pkg" {
 		cwd = filepath.Join(cwd, "..")
 	}
 
 	check := []string{
-		filepath.Join(filepath.Dir(exe), "plugin.json"),
-		filepath.Join(filepath.Dir(exe), "..", "dist", "plugin.json"),
+		filepath.Join(curProcPath, "plugin.json"),
+		filepath.Join(curProcPath, "dist", "plugin.json"),
+		filepath.Join(filepath.Dir(curProcPath), "plugin.json"),
+		filepath.Join(filepath.Dir(curProcPath), "..", "dist", "plugin.json"),
 		filepath.Join(cwd, "dist", "plugin.json"),
 		filepath.Join(cwd, "plugin.json"),
 	}
 
 	for _, path := range check {
 		if _, err := os.Stat(path); err == nil {
-			return path, nil
+			return filepath.Dir(path), nil
 		}
 	}
 
-	return exe, fmt.Errorf("can not find plugin.json in: %v", check)
+	return "", fmt.Errorf("can not find plugin.json in: %v", check)
 }
 
-func getStandaloneAddress(pluginID string, info Args) (string, error) {
-	if info.Debugger {
-		port, err := getFreePort()
-		if err != nil {
-			return "", fmt.Errorf("get free port: %w", err)
-		}
-		return fmt.Sprintf(":%d", port), nil
-	}
+// debuggedEnabled returns true if the plugin should run in debug mode.
+// It will first check the -debug flag and the name of the running process.
+func debuggerEnabled(curProcPath string) bool {
+	flag.Parse()
 
-	// Address from environment variable
-	envvar := "GF_PLUGIN_GRPC_ADDRESS_" + strings.ReplaceAll(strings.ToUpper(pluginID), "-", "_")
-	if v, ok := os.LookupEnv(envvar); ok {
-		return v, nil
+	// VsCode names the file "__debug_bin"
+	vsCodeDebug := strings.HasPrefix(filepath.Base(curProcPath), "__debug_bin")
+	// GoLand places it in:
+	//  Linux: /tmp/GoLand/___%d%(CONFIGNAME)s_pkg
+	//  Mac OS X: /private/var/folders/lx/XXX/T/GoLand/___%d%(CONFIGNAME)s_pkg
+	//  Windows: C:\Users\USER\AppData\Local\Temp\GoLand\___%d%(CONFIGNAME)s_pkg.exe
+	// Note: We also want to confirm we're not running a Go test binary
+	goLandDebug := strings.Contains(curProcPath, filepath.Join("GoLand", "___")) && filepath.Ext(curProcPath) != ".test"
```

Comment:
@xnyo this is what I mentioned

```go
&& filepath.Ext(curProcPath) != ".test"
```

😭

Reply:
Ah I see, I agree that `&& filepath.Ext(curProcPath) != ".test"` is a bit annoying, but honestly I think it's also fine :+1: